app-libs/stf/src/stf_sgx_litentry.rs

```diff
@@ -62,16 +56,26 @@ impl Stf {
 			metadata,
 			bn
 		);
-		// let parentchain_number = ita_sgx_runtime::pallet_parentchain::Pallet::<Runtime>::block_number();
+
 		ita_sgx_runtime::IdentityManagementCall::<Runtime>::link_identity {
-			who,
-			identity,
+			who: who.clone(),
+			identity: identity.clone(),
 			metadata,
 			linking_request_block: bn,
 		}
 		.dispatch_bypass_filter(ita_sgx_runtime::Origin::root())
 		.map_err(|e| StfError::Dispatch(format!("{:?}", e.error)))?;
-		// TODO: generate challenge code
+
+		// generate challenge code
+		let code = generate_challenge_code();
```

Comment:
I think we should return the `challenge_code` and let the parachain know that the `challenge_code` has been generated

Reply:
Indeed, thanks!